### DISCUSSION TITLE
TWAIN ERROR : Bummer

### DISCUSSION THREAD

**Discussion by Nidkfr** [cyanfish/naps2#429] [Support]
Hello,

   When run 1 scan of 1 page , all is good. 
   But il run another scan in same programme session i have this error : **TWAIN ERROR : Bummer**.
   If i restart my app , 1 page is ok , if run another scan fail with error.
   I try dispose scannercontext, re-retreive scan , wait 1 minute , when run second scan , i have this error.
   Juste first scan work ? have a idea  for call method Scan many times ?

**Comment by cyanfish** [comment]
Are you use a worker for TWAIN scanning? If not, follow [these instructions](https://github.com/cyanfish/naps2/blob/master/NAPS2.Sdk.Samples/TwainSample.cs) to set it up, i.e. installing NAPS2.Sdk.Worker.Win32 and calling `scanningContext.SetUpWin32Worker();`.

**Reply by Nidkfr** [reply]
if I understood correctly, you have to use this feature if you are not in 32 bits. Except my tests are done from a 32-bit application.
In addition, this feature launches a worker that launches the scan then stops, and therefore it does not have the same problem as me. To resolve this situation, I also created a worker (because the one proposed gives me problems on where the exe is).
But I would find it more efficient if I am not forced to restart an exe each time? I think Dispose does not free all the resources?

**Reply by cyanfish** [reply]
Yes, that would be ideal to not need a worker process. But the problem is in the TWAIN driver provided by your scanner manufacturer, not in NAPS2.Sdk.

If you really don't want a worker process you can experiment with using [NTwain](https://github.com/soukoku/ntwain) instead of NAPS2.Sdk. Depending on the exact nature of the driver problem it might be possible to make it work.
